src/AcceptanceTests/When_customizing_outgoing_messages.cs

```diff
@@ -0,0 +1,71 @@
+namespace NServiceBus.Transport.RabbitMQ.AcceptanceTests
+{
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using Azure.Messaging.ServiceBus;
+    using NServiceBus.AcceptanceTests;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NUnit.Framework;
+
+    class When_customizing_outgoing_messages : NServiceBusAcceptanceTest
+    {
+        const string TestCaseKey = "0192c3ad-8ab2-77a0-8a92-2be53f062e06";
+        const string TestCaseValue = "0192c3ad-c1d2-71b3-9b8e-ed0d771e8785";
+
+        [Test]
+        public async Task Should_receive_custom_set_value()
+        {
+            var scenario = await Scenario.Define<Context>()
+                .WithEndpoint<Receiver>(b => b.When((bus, c) => bus.SendLocal(new Message())))
+                .Done(c => c.MessageReceived)
+                .Run();
+
+            Assert.That(scenario.ReceivedMessage.ApplicationProperties[TestCaseKey], Is.EqualTo(TestCaseValue));
+        }
+
+        public class Receiver : EndpointConfigurationBuilder
+        {
+            public Receiver()
+            {
+                EndpointSetup<DefaultServer>(endpointConfiguration =>
+                {
+                    var t = (AzureServiceBusTransport)endpointConfiguration.ConfigureTransport();
+                    t.OutgoingNativeMessageCustomization =
+                        (operation, message) =>
+                        {
+                            message.ApplicationProperties[TestCaseKey] = TestCaseValue;
+                        };
+                });
+            }
+
+            class MyEventHandler : IHandleMessages<Message>
+            {
+                Context testContext;
+
+                public MyEventHandler(Context testContext)
+                {
+                    this.testContext = testContext;
+                }
+
+                public Task Handle(Message message, IMessageHandlerContext context)
+                {
+                    testContext.ReceivedMessage = context.Extensions.Get<ServiceBusReceivedMessage>();
```

Comment:
Funny enough the application properties are headers. So you could simply check extract the context.Headers here in case you want. Assuming that would it better to set something like subject?

Reply:
I didn't touch other properties to avoid conflict with transport used properties. Specifically because it now uses an AT but I'll test with subject. Didn't know that property yet.

Reply:
Applied on branch